server/src/main/java/org/eclipse/lsp/cobol/core/visitor/VariableDefinitionDelegate.java

```diff
@@ -264,6 +265,45 @@ public void notifySectionChanged() {
     return new ResultWithErrors<>(new ArrayList<>(variables), new ArrayList<>(errors));
   }
 
+  /**
+   * Handle and process REDEFINES clause
+   * @param context is a ParserRuleContext context
+   */
+  public void handleRedefine(DataRedefinesClauseContext context) {
+    String redefinesName = VisitorHelper.getName(context.dataName());
+    Locality locality = positions.get(context.dataName().getStart());
+
+    if (locality != null) {
+      boolean foundRoot = false;
+      Iterator<Variable> iterator = variables.iterator();
+      if (iterator.hasNext()) {
+        Variable currentVariable = iterator.next();
+
+        while (iterator.hasNext()) {
```

Comment:
A variable redefines the previous one, so it would be faster to check them from the end

Reply:
it is Deque, so the order is from the last to the first already